addons/mail/static/src/messaging/entity/thread/thread.js

```diff
@@ -975,7 +1002,42 @@ function ThreadFactory({ Entity }) {
                 'name',
             ],
         }),
+        /**
+         * Determine the fold state of the channel on the web client.
+         *
+         * If there is a pending fold state change, it is immediately applied on
+         * the interface to avoid a feeling of unresponsiveness.
+         *
+         * Otherwise the last known fold state of the server is used.
+         *
+         * This field must be considered read only.
+         */
         foldState: attr({
+            compute: '_computeFoldState',
+            dependencies: [
+                'foldStatePending',
+                'foldStateServer',
+            ],
+        }),
+        /**
+         * Determine if there is a pending fold state change, which is a change
+         * of fold state requested by the client but not yet confirmed by the
+         * server.
+         *
+         * This field can be updated to immediately change the fold state on the
+         * interface and to notify the server of the new state.
+         */
+        foldStatePending: attr(),
+        /**
+         * Determine the last fold state known by the server, which is the fold
+         * fold state displayed after initialization or when the last pending
+         * fold state change was confirmed by the server.
+         *
+         * This field should be considered read only in most situations. Only
+         * the code handling fold state change from the server should typically
+         * update it.
+         */
+        foldStateServer: attr({
```

Comment:
I would prefer too but then they are not together in the field definition due to alphabetical which is annoying.

I don't want to waste time on this, just let me know final decision considering this remark and I'll apply it.

Reply:
I get they are not close in the definition due to alphabetical order, but it's confusing to read `foldStateServer` and in reality it means `serverFoldState`, so that's a reason to rename to reflect it's intent correctly.

The problem of fields of same concern not being close enough is not a motivation to name them in a misleading way (`foldStateServer` could model the server containing the fold state? I dunno, it seems super stupid for us but when code becomes even more complex this confusion will happens, even more if we do same bad naming elsewhere). It's complicated to solve, hence several implementation techniques and paradigms (inheritance, composition, mixins, "component hooks", Aspect-Oriented Programming, Context-Oriented Programming, Feature-Oriented Programming, etc...), each have pros and cons.

I think this is a lesser problem than the bad naming one.

Reply:
Ok, renamed